packages/fx-core/src/common/template-utils/templatesActions.ts

```diff
@@ -215,6 +216,24 @@ export async function scaffoldFromTemplates(
   context: ScaffoldContext,
   actions: ScaffoldAction[] = defaultActionSeq
 ): Promise<void> {
+  if (!context.group) {
+    throw new Error(missKeyErrorInfo("group"));
+  }
+
+  if (!context.lang) {
+    throw new Error(missKeyErrorInfo("lang"));
+  }
+
+  if (!context.scenario) {
+    throw new Error(missKeyErrorInfo("scenario"));
+  }
+  // To track code templates usage.
+  sendTelemetryEvent(Component.core, TelemetryEvent.ScaffoldFromTemplatesStart, {
```

Comment:
Do we really need start event? Maybe we can discuss offline.

Reply:
I used start event to track all scaffold intentions in the very beginning of scaffold. You know, these intentions may not be all successful, so I got another telemetry in the end with fallback property.